src/librustdoc/formats/cache.rs

```diff
@@ -244,10 +245,17 @@ impl<'a, 'tcx> DocFolder for CacheBuilder<'a, 'tcx> {
             }
         }
 
+        let tcx = self.tcx;
         // Propagate a trait method's documentation to all implementors of the
         // trait.
         if let clean::TraitItem(ref t) = *item.kind {
-            self.cache.traits.entry(item.def_id).or_insert_with(|| t.clone());
+            self.cache.traits.entry(item.def_id).or_insert_with(|| clean::TraitWithExtraInfo {
+                trait_: t.clone(),
+                is_spotlight: clean::utils::has_doc_flag(
+                    tcx.get_attrs(item.def_id),
+                    sym::spotlight,
+                ),
+            });
```

Comment:
Not necessary to change now since this code was here before, but couldn't we simplify this code and avoid the `Entry` API since we aren't using the returned `&mut V`?

Reply:
I added a FIXME.

Reply:
I disagree with the FIXME. Entry is more efficient because it only has to calculate the hash once (compared to `.contains`, which has to hash twice).